src/AggregateFunctions/AggregateFunctionCramersV.h

```diff
@@ -0,0 +1,198 @@
+#pragma once
+
+#include <AggregateFunctions/IAggregateFunction.h>
+#include <Columns/ColumnVector.h>
+#include <Columns/ColumnTuple.h>
+#include <Common/assert_cast.h>
+#include <Common/FieldVisitors.h>
+#include <Core/Types.h>
+#include <DataTypes/DataTypesDecimal.h>
+#include <DataTypes/DataTypeNullable.h>
+#include <DataTypes/DataTypesNumber.h>
+#include <DataTypes/DataTypeTuple.h>
+#include <IO/ReadHelpers.h>
+#include <IO/WriteHelpers.h>
+#include <Common/HashTable/HashMap.h>
+#include <AggregateFunctions/UniqVariadicHash.h>
+#include <limits>
+#include <cmath>
+
+#include <type_traits>
+
+namespace DB
+{
+
+
+    struct AggregateFunctionCramersVData
+    {
+        size_t cur_size = 0;
+        HashMap<UInt64, UInt64> n_i;
+        HashMap<UInt64, UInt64> n_j;
+        HashMap<UInt128, UInt64> pairs;
+
+
+        void add(UInt64 hash1, UInt64 hash2)
+        {
+            cur_size += 1;
+            n_i[hash1] += 1;
+            n_j[hash2] += 1;
+
+            UInt128 hash_pair = hash1 | (static_cast<UInt128>(hash2) << 64);
+            pairs[hash_pair] += 1;
+
+        }
+
+        void merge(const AggregateFunctionCramersVData &other)
+        {
+            cur_size += other.cur_size;
+
+            for (const auto& pair : other.n_i) {
+                UInt64 hash1 = pair.getKey();
+                UInt64 count = pair.getMapped();
+                n_i[hash1] += count;
+            }
+            for (const auto& pair : other.n_j) {
+                UInt64 hash1 = pair.getKey();
+                UInt64 count = pair.getMapped();
+                n_j[hash1] += count;
+            }
+            for (const auto& pair : other.pairs) {
+                UInt128 hash1 = pair.getKey();
+                UInt64 count = pair.getMapped();
+                pairs[hash1] += count;
+            }
+        }
+
+        void serialize(WriteBuffer &buf) const
+        {
+            writeBinary(cur_size, buf);
+            n_i.write(buf);
+            n_j.write(buf);
+            pairs.write(buf);
```

Comment:
We can lower the amount of data passed over network if we only serialize pairs and reconstruct n_i and n_j during deserialization.